### PR TITLE
Fix bug in pca metrics warnings

### LINKED ISSUE #1873: f strings not joined before passed to warnings in pca_metrics
On line 466 in the pca_metrics file (https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/qualitymetrics/pca_metrics.py), the following lines result in an error. It seems like the f strings are getting interpreted as separate arguments (one of which should be an integer) rather than joined and interpreted as a single argument. There are a few other places in this file with a similar issue. Removing commands fixed the issue.

`TypeError: 'str' object cannot be interpreted as an integer`

`    if n_spikes_all_units[this_unit_id] < min_spikes:
        warnings.warn(
            f"Warning: unit {this_unit_id} has fewer spikes than ",
            f"specified by `min_spikes` ({min_spikes}); ",
            f"returning NaN as the quality metric...",
        )
        return np.nan, np.nan`

### PR DESCRIPTION
**Pull request by alejoe91**
Fix #1873

### CODE REVIEW

File: src/spikeinterface/qualitymetrics/pca_metrics.py [hunk 1]
```diff
@@ -466,15 +466,14 @@ def nearest_neighbors_isolation(
...
-            f"returning NaN as the quality metric...",
+            f"Warning: unit {this_unit_id} has fewer spikes than specified by `min_spikes` "
+            f"({min_spikes}); returning NaN as the quality metric..."
         )
         return np.nan, np.nan
     elif fr_all_units[this_unit_id] < min_fr:
         warnings.warn(
-            f"Warning: unit {this_unit_id} has a firing rate ",
-            f"below the specified `min_fr` ({min_fr}Hz); " f"returning NaN as the quality metric...",
+            f"Warning: unit {this_uit_id} has a firing rate below the specified `min_fr` "
```
Comment (h-mayorquin) [inline src/spikeinterface/qualitymetrics/pca_metrics.py:475]: Should said unit not uid, right?

File: src/spikeinterface/qualitymetrics/pca_metrics.py [hunk 2]
```diff
@@ -652,15 +651,14 @@ def nearest_neighbors_noise_overlap(
...
+            f"Warning: unit {this_unit_id} has fewer spikes than specified by `min_spikes` "
+            f"({min_spikes}); returning NaN as the quality metric..."
         )
         return np.nan
     elif fr_all_units[this_unit_id] < min_fr:
         warnings.warn(
-            f"Warning: unit {this_unit_id} has a firing rate ",
-            f"below the specified `min_fr` ({min_fr}Hz); " f"returning NaN as the quality metric...",
+            f"Warning: unit {this_unit_id} has a firing rate below the specified `min_fr` "
+            f"({min_fr}Hz); returning NaN as the quality metric...",
```
Comment (h-mayorquin) [inline src/spikeinterface/qualitymetrics/pca_metrics.py:661]:
```suggestion
            f"({min_fr} Hz); returning NaN as the quality metric...",
```

File: src/spikeinterface/qualitymetrics/pca_metrics.py [hunk 3]
```diff
@@ -466,15 +466,14 @@ def nearest_neighbors_isolation(
...
+            f"Warning: unit {this_unit_id} has fewer spikes than specified by `min_spikes` "
+            f"({min_spikes}); returning NaN as the quality metric..."
         )
         return np.nan, np.nan
     elif fr_all_units[this_unit_id] < min_fr:
         warnings.warn(
-            f"Warning: unit {this_unit_id} has a firing rate ",
-            f"below the specified `min_fr` ({min_fr}Hz); " f"returning NaN as the quality metric...",
+            f"Warning: unit {this_uit_id} has a firing rate below the specified `min_fr` "
+            f"({min_fr}Hz); returning NaN as the quality metric..."
```
Comment (h-mayorquin) [inline src/spikeinterface/qualitymetrics/pca_metrics.py:476]:
```suggestion
            f"({min_fr} Hz); returning NaN as the quality metric..."
```

**Review by h-mayorquin — APPROVED** [review]
I think that the `Warning:` at the beginning of the warning messages is redundant:

```
import warnings 
warnings.warn("This is a warning")
Outputs:
UserWarning: This is a warning
  warnings.warn("This is a warning")
```

File: src/spikeinterface/qualitymetrics/pca_metrics.py (hunk 1 above)
Comment (alejoe91) [inline src/spikeinterface/qualitymetrics/pca_metrics.py:475]:
```suggestion
            f"Warning: unit {this_unit_id} has a firing rate below the specified `min_fr` "
```

**Comment by alejoe91** [comment]
> I think that the `Warning:` at the beginning of the warning messages is redundant:
> 
> ```
> import warnings 
> warnings.warn("This is a warning")
> Outputs:
> UserWarning: This is a warning
>   warnings.warn("This is a warning")
> ```

done!

**Pull request merged**